lib/src/view/tournament/tournament_screen.dart

```diff
@@ -0,0 +1,489 @@
+                const SizedBox(width: 5),
+              ],
+              Text('${player.score}'),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _StandingControls extends ConsumerWidget {
+  const _StandingControls({required this.state});
+
+  final TournamentState state;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        IconButton(
+          onPressed:
+              state.hasPreviousPage
+                  ? ref.read(tournamentControllerProvider(state.id).notifier).loadFirstStandingsPage
+                  : null,
+          icon: const Icon(Icons.first_page),
+        ),
+        IconButton(
+          onPressed:
+              state.hasPreviousPage
+                  ? ref
+                      .read(tournamentControllerProvider(state.id).notifier)
+                      .loadPreviousStandingsPage
+                  : null,
+          icon: const Icon(Icons.skip_previous),
+        ),
+        Text(
+          '${state.firstRankOfPage}-${min(state.firstRankOfPage + kStandingsPageSize - 1, state.tournament.nbPlayers)} / ${state.tournament.nbPlayers}',
+        ),
+        IconButton(
+          onPressed:
+              state.hasNextPage
+                  ? ref.read(tournamentControllerProvider(state.id).notifier).loadNextStandingsPage
+                  : null,
+          icon: const Icon(Icons.skip_next),
+        ),
+        IconButton(
+          onPressed:
+              state.hasNextPage
+                  ? ref.read(tournamentControllerProvider(state.id).notifier).loadLastStandingsPage
+                  : null,
+          icon: const Icon(Icons.last_page),
+        ),
+        if (state.tournament.me != null)
+          IconButton(
+            onPressed: ref.read(tournamentControllerProvider(state.id).notifier).jumpToMyPage,
+            icon: const Icon(LichessIcons.target),
+          ),
+      ],
+    );
+  }
+}
+
+class _Verdicts extends ConsumerWidget {
+  const _Verdicts(this.verdicts);
+
+  final Verdicts verdicts;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isLoggedIn = ref.watch(authSessionProvider)?.user.id != null;
+
+    if (verdicts.list.isEmpty) {
+      return const SizedBox.shrink();
+    }
+
+    return Row(
+      children: [
+        Icon(
+          isLoggedIn && verdicts.accepted ? Icons.check : Icons.lock,
+          color:
+              isLoggedIn
+                  ? verdicts.accepted
+                      ? LichessColors.good
+                      : LichessColors.error
+                  : null,
+          size: 30,
+        ),
+        const SizedBox(width: 10),
+        Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: verdicts.list
+              .whereNot((verdict) => verdict.condition == 'Bot players are not allowed')
```

Comment:
Could be refactored with a list comprehension and a `if` inside, which is usually more readable (less indented).